src/components/interaction-designer/toolbar/BlockErrorsExpandable.vue

```diff
@@ -57,12 +59,21 @@ export default class BlockErrorsExpandable extends mixins(Lang) {
 
   get errorsToShow(): ErrorObject[] {
     return this.isExpanded
-      ? this.status.ajvErrors ?? []
-      : this.status.ajvErrors?.slice(0, DEFAULT_LIST_SIZE) ?? []
+      ? this.allErrors ?? []
+      : this.allErrors?.slice(0, DEFAULT_LIST_SIZE) ?? []
+  }
+
+  /**
+   * get allErrors to consider errors from:
+   * - block validation
+   * - resource validation
+   */
+  get allErrors(): ErrorObject[] {
+    return union(this.status.ajvErrors, this.resourceValidationStatuses.ajvErrors);
```

Comment:
The block validation errors are passed down from the parent component as a 'status' _prop_, whereas resource validation errors are read from the _vuex store_.
Do you think it would make sense to unify this and read the block validation errors from the store as well?

P.S. I think I was the one who did it via the props, ignoring the global store approach. My bad 😄

Reply:
@bulatgab, I think we don't really need to always get data from vuex (or always send them from parent).
The key point here: As`BlockErrorsExpandable.vue` is handling errors for **01 block**. So, if I change `resourceValidationStatuses` into `resourceValidationStatusesForCurrentBlockValidationStatus`, that would be less confusing. What do you think ?

Reply:
Okay, a mixed approach then.
Regarding the `resourceValidationStatuses` - personally, it wasn't confusing to me, but I'm okay with the new name.